src/RemoteRequest/WpHttpRemoteGetRequest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * Class WpHttpRemoteGetRequest.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\RemoteRequest;
+
+use AmpProject\Exception\FailedToGetFromRemoteUrl;
+use AmpProject\RemoteGetRequest;
+use AmpProject\RemoteRequest\RemoteGetRequestResponse;
+use AmpProject\Response;
+use Traversable;
+use WP_Error;
+use WP_Http;
+
+/**
+ * Remote request transport using the WordPress WP_Http abstraction layer.
+ *
+ * @package AmpProject\AmpWP
+ */
+final class WpHttpRemoteGetRequest implements RemoteGetRequest {
+
+	/**
+	 * Default timeout value to use in seconds.
+	 *
+	 * @var int
+	 */
+	const DEFAULT_TIMEOUT = 5;
+
+	/**
+	 * Default number of retry attempts to do.
+	 *
+	 * @var int
+	 */
+	const DEFAULT_RETRIES = 2;
+
+	/**
+	 * List of HTTP status codes that are worth retrying for.
+	 *
+	 * @var int[]
+	 */
+	const RETRYABLE_STATUS_CODES = [
+		WP_Http::REQUEST_TIMEOUT,
+		WP_Http::LOCKED,
+		WP_Http::TOO_MANY_REQUESTS,
+		WP_Http::INTERNAL_SERVER_ERROR,
+		WP_Http::SERVICE_UNAVAILABLE,
+		WP_Http::GATEWAY_TIMEOUT,
+	];
+
+	/**
+	 * Whether to verify SSL certificates or not.
+	 *
+	 * @var boolean
+	 */
+	private $ssl_verify;
+
+	/**
+	 * Timeout value to use in seconds.
+	 *
+	 * @var int
+	 */
+	private $timeout;
+
+	/**
+	 * Number of retry attempts to do for an error that is worth retrying.
+	 *
+	 * @var int
+	 */
+	private $retries;
+
+	/**
+	 * Instance of the WordPress HTTP abstraction object.
+	 *
+	 * @var WP_Http
+	 */
+	private $wp_http;
+
+	/**
+	 * Instantiate a WpHttpRemoteGetRequest object.
+	 *
+	 * @param bool $ssl_verify Optional. Whether to verify SSL certificates. Defaults to true.
+	 * @param int  $timeout    Optional. Timeout value to use in seconds. Defaults to 10.
+	 * @param int  $retries    Optional. Number of retry attempts to do if a status code was thrown that is worth
+	 *                        retrying. Defaults to 2.
+	 */
+	public function __construct( $ssl_verify = true, $timeout = self::DEFAULT_TIMEOUT, $retries = self::DEFAULT_RETRIES ) {
+		if ( ! is_int( $timeout ) || $timeout < 0 ) {
+			$timeout = self::DEFAULT_TIMEOUT;
+		}
+
+		if ( ! is_int( $retries ) || $retries < 0 ) {
+			$retries = self::DEFAULT_RETRIES;
+		}
+
+		$this->ssl_verify = $ssl_verify;
+		$this->timeout    = $timeout;
+		$this->retries    = $retries;
+
+		$this->wp_http = _wp_http_get_object();
```

Comment:
Question: `_wp_http_get_object()` is marked as `@private`. While WordPress themes and plugins rarely respect such things, there's a chance in the future it could go away, no? Why not use `wp_remote_get()` directly?

Reply:
Resolved via https://github.com/ampproject/amp-wp/pull/4404/commits/163228630ecaf26bfd587807f2d1e1e8917a0f0c